keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactVersionLinks.kt

```diff
@@ -24,13 +24,15 @@ fun generateCompareLink(scmInfo: ScmInfo, version1: PublishedArtifact?, version2
 
 //Generating a SCM compare link between source (new version) and target (old version) versions (the order matter!)
 private fun generateCompareLink(scmInfo: ScmInfo, newerGitMetadata: GitMetadata?, olderGitMetadata: GitMetadata?): String? {
-  val baseScmUrl = newerGitMetadata?.commitInfo?.link?.let { getScmBaseLink(scmInfo, it) }
-  return if (baseScmUrl != null && olderGitMetadata != null && !(olderGitMetadata.commitInfo?.sha.isNullOrEmpty())) {
-    "$baseScmUrl/projects/${newerGitMetadata.project}/repos/${newerGitMetadata.repo?.name}/compare/commits?" +
+  var commitLink = newerGitMetadata?.commitInfo?.link ?: return null
```

Comment:
how about something like this, to make it a bit clearer? 
```suggestion
  var commitLink = newerGitMetadata?.commitInfo?.link ?: return null
    return when {
    "stash" in commitLink -> {
      
      val baseScmUrl =  getScmBaseLink(scmInfo, commitLink)
      if (baseScmUrl != null && olderGitMetadata != null && !(olderGitMetadata.commitInfo?.sha.isNullOrEmpty())) {
        "$baseScmUrl/projects/${newerGitMetadata.project}/repos/${newerGitMetadata.repo?.name}/compare/commits?" +
          "targetBranch=${olderGitMetadata.commitInfo?.sha}&sourceBranch=${newerGitMetadata.commitInfo?.sha}"
      } else null
    }
    "github" in commitLink ->
      commitLink
    else ->  throw UnsupportedScmType(message = "Stash & Git are currently the only supported SCM type")
  }
```

Reply:
I generally try to avoid `if-else` blocks wherever I can, but then again I am very new to Kotlin, so if this is the better style in Kotlin, I am happy to change it.